test/unit/test_compatability.py

```diff
@@ -22,13 +22,12 @@
 
 
 class TestWarnDeprecatedPython:
-    def test_happy_version(self):
+    def test_happy_version(self, recwarn):
```

Comment:
Question: Sorry, I do not follow.
Why is this argument necessary?

Is this some magic incantation to tell pytest to record warnings?
And then assert that none were recorded?

Reply:
Yes, these are called fixtures in pytest, see https://docs.pytest.org/en/8.2.x/how-to/capture-warnings.html#recwarn

This change was necessary because `pytest.warns(None)` is not allowed anymore.